library/network_connections.py

```diff
@@ -2639,6 +2642,519 @@ def run_action_down(self, idx):
 ###############################################################################
 
 
+class Cmd_nm_offline(Cmd):
+    def __init__(self, **kwargs):
+        Cmd.__init__(self, **kwargs)
+        self.validate_one_type = ArgValidator_ListConnections.VALIDATE_ONE_MODE_NM
+        self._checkpoint = None
+
+    def profile_path(self, name):
+        return os.path.join(
+            "/etc/NetworkManager/system-connections", name + ".nmconnection"
+        )
+
+    def run_prepare(self):
+        # we can't check any hardware or runtime status, we can just trust the input
+        pass
+
+    # mirror Cmd_nm.connection_create()
+    def connection_create(self, connections, idx):
+        connection = connections[idx]
+        # global/type independent arguments
+        if "nm.uuid" not in connection:
+            connection["nm.uuid"] = Util.create_uuid()
+        if "nm.exists" not in connection:
+            connection["nm.exists"] = True
+        argv = [
+            "nmcli",
```

Comment:
Instead of spawning nmcli, did you consider using libnm to create the connection in memory (similarly to `NMUtil.connection_create`) and then using [nm_keyfile_write()](https://networkmanager.dev/docs/libnm/latest/libnm-nm-keyfile.html)  to write it to disk? Probably that would allow some code reuse.